dominion/player.py

```diff
@@ -17,36 +18,36 @@ class Player:
 
     def __init__(self, deck: Deck):
         self.deck = deck
-        self.player_id = base64.b64encode(
-            str(time.time()).zfill(24).encode()
-        ).decode() + "".join(
-            random.choice(
-                "1234567890qwertyuiopasdfghjklzxcvbnmQWERTYUIOPASDFGHJKLZXCVBNM"
-            )
-            for _ in range(10)
-        )
+        self.player_id = uuid.uuid4()
 
     def action_phase(self):
-        raise NotImplementedError
```

Comment:
This action phase function is not supposed be implemented by the Base player class, turn logic happens in Game.